### DISCUSSION TITLE
Possible to prevent battery discharge using this integration?

### DISCUSSION THREAD

**Discussion by andrewmooreio** [nathanmarlor/foxess_modbus#173] [Q&A] [answered]
Hi all,

Just wondering if it's possible to prevent battery discharge using this integration.

Octopus have lowered their electricity overnight rates for the Go and Intelligent tariffs, making it more economical to heat my hot water tank using electricity instead of gas. However, I can only do this if I can prevent my batteries from discharging between 00:30-04:30. Otherwise, my batteries will discharge, and I'll be running the house off the grid until the sun comes up.

Thanks in advance.

**Comment by canton7** [comment, marked as answer]
Sure, you can set charge periods using this integration, the same as in the foxess app.

I'm still working on a nice dashboard card, but you can use the services. Go to Developer Tools -> Services and select FoxESS - Modbus: Update Charge Period. Enable "Enable force charge" but disable "Enable charge from grid", to prevent the battery from discharging but stop it from charging from the grid. Then "Call Service". You'll see the results under the "Time Period X" sensors.

![image](https://user-images.githubusercontent.com/568104/233780725-f40512f5-595f-4e87-b879-62a2a640e4b3.png)

**Comment by andrewmooreio** [comment]
Perfect, thank you. I wasn't 100% clear on what “Enable force charge” did.

**Reply by canton7** [reply]
Yeah, I copied the same wording that FoxESS use in their app / website, but that's not hugely clear...

**Reply by canton7** [reply]
If you've got a proposal for some better wording here, I'll happily take it!
